### PR TITLE
Guard against GCS rate limits

### PR DESCRIPTION
**Pull request by blowmage**
This PR guards against rate limit error when trying to create or delete buckets. This is a frequent cause of failing acceptance tests.

### CODE REVIEW

File: google-cloud-bigquery/acceptance/bigquery_helper.rb [hunk 1]
```diff
@@ -78,6 +78,20 @@ def self.run_one_method klass, method_name, reporter
   end
 end
 
+def safe_gcs_execute retries: 20, delay: 2
+  current_retries = 0
+  loop do
+    begin
+      return yield
+    rescue Google::Cloud::ResourceExhaustedError
```
Comment (quartzmo) [inline google-cloud-bigquery/acceptance/bigquery_helper.rb]: add `=> e` to end of this line.

File: google-cloud-speech/acceptance/speech_helper.rb [hunk 2]
```diff
@@ -77,6 +77,20 @@ def self.run_one_method klass, method_name, reporter
   end
 end
 
+def safe_gcs_execute retries: 20, delay: 2
+  current_retries = 0
+  loop do
+    begin
+      return yield
+    rescue Google::Cloud::ResourceExhaustedError
```
Comment (quartzmo) [inline google-cloud-speech/acceptance/speech_helper.rb]: add `=> e` to end of this line.

File: google-cloud-storage/acceptance/storage_helper.rb [hunk 3]
```diff
@@ -96,6 +96,20 @@ def try_with_backoff msg = nil, limit: 10
   end
 end
 
+def safe_gcs_execute retries: 20, delay: 2
+  current_retries = 0
+  loop do
+    begin
+      return yield
+    rescue Google::Cloud::ResourceExhaustedError
```
Comment (quartzmo) [inline google-cloud-storage/acceptance/storage_helper.rb]: add `=> e` to end of this line.

File: google-cloud-vision/acceptance/vision_helper.rb [hunk 4]
```diff
@@ -86,6 +86,20 @@ def self.run_one_method klass, method_name, reporter
   end
 end
 
+def safe_gcs_execute retries: 20, delay: 2
+  current_retries = 0
+  loop do
+    begin
+      return yield
+    rescue Google::Cloud::ResourceExhaustedError
```
Comment (quartzmo) [inline google-cloud-vision/acceptance/vision_helper.rb]: add `=> e` to end of this line.

Comment (blowmage) [reply google-cloud-vision/acceptance/vision_helper.rb]: Fixed.

File: google-cloud-bigquery/acceptance/bigquery_helper.rb (hunk 1 above)
Comment (blowmage) [reply google-cloud-bigquery/acceptance/bigquery_helper.rb]: Fixed.

**Comment by blowmage** [comment]
[refs #2088]

**Pull request merged**